### DISCUSSION TITLE
Add ISR Support for Caching

### DISCUSSION THREAD

**Discussion by jdgamble555** [solidjs/solid#1298] [Ideas]
Rich Harris has said he will add ISR to SvelteKit after version one.  I think this feature is huge in the community.  There are a lot of NextJS users who don't like React, but use NextJS simply for the ISR Support.

Allowing page caching would be huge for SolidJS, and I believe would bring in more people into the community who don't want to use React, which is important. 

All ISR is, is easily adaptation of the [cache-control header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) with in your app on edge functions.

I believe this could be easily done with much reward.

J

**Comment by lxsmnsyc** [comment]
I think this should be best asked at https://github.com/solidjs/solid-start

on the ISR, I've made a proposal to include a `cache` integration for the purpose of ISR. The `cache` integration would accept an interface similar to a KV store, except that the implementation may vary on the integration, which opens to a lot of opportunities like using Cloudflare's KV store, Redis, File-based caching, In-memory, etc.

**Comment by trompx** [comment]
Currently using next.js but I am also looking for another framework doing ISR because I'd like to avoid the vercel vendor lock in, where economic incentives are shaping the framework more and more, leading to a worst DX as it is a pain to deploy outside of their platform (which is crazy expensive when you need to scale).

**Comment by maxmilton** [comment]
TBH, reading this I had no idea what ISR is. For clarity, ISR is "incremental static regeneration". It seems to be a build strategy coming from Next.js. Here's some info:

- <https://vercel.com/docs/concepts/incremental-static-regeneration/overview>
- <https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration>

It's not something I want personally but an interesting idea nevertheless.
